### DISCUSSION TITLE
Optogenetics and access to Harp software with Bash

### DISCUSSION THREAD

**Discussion by AstridNicolini15** [harp-tech/protocol#86] [Q&A]
Hello, 

We have a setup controlled by Harp and have some questions : 

1)	We would like to add an optogenetics laser to our setup. Have you ever done that, and can recommend a procedure ? Otherwise, we are thinking of using Harp PMW outlet to send triggers to the optogenetics LED driver. Do you think its doable ?

2)	We want to automatize some maintenance task of our setup using Bash. To do so, we would like to change parameters (such as the valve opening duration -poke0-) but without having to open Harp software and changing it manually. Do you know if its possible to directly access and modify Harp code ? If yes, how ? 

Thank you very much

**Comment by filcarv** [comment]
👋
1. Some devices have PWM capabilities, like the Harp Behavior, the Harp Output Expander, and the Harp Multi Pwm Output. This latest one has more options, but if you have any of the first two on this list, you should be fine.
2. The Harp devices' parameters can be changed in Bonsai. If you are able to send this info to Bonsai using Bash or have a file that Bonsai can read at startup, you should be able to configure these parameters.
